### PR TITLE
[NEXUS-8413] apply config-facet changes to use bean-mapping+validation

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8413

Should wait for https://github.com/sonatype/nexus-oss/pull/1190 to merge and reconcile before merging

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/proxy/ProxyFacetSupport.java [hunk 1]
```diff
@@ -53,44 +56,68 @@
...
-
-  private URI remoteUrl;
+  @VisibleForTesting
+  static final String CONFIG_KEY = "proxy";
+
+  @VisibleForTesting
+  static class Config
+  {
+    // TODO: Normalize URI ends with "/"?
+    @NotEmpty
```
Comment (adreghiciu) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/proxy/ProxyFacetSupport.java]: Let's not forget to normalize URI

**Comment by adreghiciu** [comment]
+1

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/proxy/ProxyFacetSupport.java (hunk 1 above)
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/proxy/ProxyFacetSupport.java]: I won't forget :-P  Just trying to decide if this is should be done after configuration is loaded, or with @JsonProperty and/or if the "remoteUrl" here should be a String and converted to URI with normalization.

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/ConfigurableViewFacet.java [hunk 2]
```diff
@@ -43,6 +43,8 @@ public void configure(final Router router) {
     this.router = router;
   }
 
+  // FIXME: Move 'online' to core repository property
```
Comment (kellyrob99) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/ConfigurableViewFacet.java]: This property came up when dealing with the UI as well, and I'm not sure what the intent for it is. https://github.com/sonatype/nexus-oss/pull/1183#discussion_r28408713

File: components/nexus-repository/src/test/java/org/sonatype/nexus/repository/httpclient/AuthenticationConfigDeserializerTest.groovy [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * Tests for {@link AuthenticationConfigDeserializer}.
+ */
+class AuthenticationConfigDeserializerTest
+    extends TestSupport
+{
+  private ObjectMapper objectMapper
+
+  @Before
```
Comment (kellyrob99) [inline components/nexus-repository/src/test/java/org/sonatype/nexus/repository/httpclient/AuthenticationConfigDeserializerTest.groovy]: Minor but you should be able to skip the before block and just instantiate as a class member

Comment (jdillon) [inline components/nexus-repository/src/test/java/org/sonatype/nexus/repository/httpclient/AuthenticationConfigDeserializerTest.groovy]: yes I know, but I prefer to use the framework

**Comment by kellyrob99** [comment]
+1 and I will plan on syncing any property name/location changes on #1183

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF173-1 :white_check_mark:

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/ConfigurableViewFacet.java (hunk 2 above)
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/ConfigurableViewFacet.java]: to allow the repository to under go admin shit but not allow users to access it

**Comment by adreghiciu** [comment]
+1

**Comment by jdillon** [comment]
Running into this now: https://gist.github.com/jdillon/4bd03fe3ab5e3a8d947b

Looks like it could be classloading issue after validation refactor.

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF173-3 :white_check_mark: 

... but still has some core issues

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF173-5 :white_check_mark:

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**